core/src/main/java/org/testcontainers/containers/BaseContainerDef.java

```diff
@@ -0,0 +1,332 @@
+        this.envVars.clear();
+        this.envVars.putAll(envVars);
+    }
+
+    protected void addEnvVars(Map<String, String> envVars) {
+        this.envVars.putAll(envVars);
+    }
+
+    protected void addEnvVar(String key, String value) {
+        this.envVars.put(key, value);
+    }
+
+    public String[] getEntrypoint() {
+        return Arrays.copyOf(this.entrypoint, this.entrypoint.length);
+    }
+
+    protected void setEntrypoint(String... entrypoint) {
+        this.entrypoint = entrypoint;
+    }
+
+    public String[] getCommand() {
+        return Arrays.copyOf(this.command, this.command.length);
+    }
+
+    protected void setCommand(String... command) {
+        this.command = command;
+    }
+
+    protected void setNetwork(Network network) {
+        this.network = network;
+    }
+
+    public Set<String> getNetworkAliases() {
+        return new LinkedHashSet<>(this.networkAliases);
+    }
+
+    protected void setNetworkAliases(Set<String> aliases) {
+        this.networkAliases.clear();
+        this.networkAliases.addAll(aliases);
+    }
+
+    protected void addNetworkAliases(String... aliases) {
+        this.networkAliases.addAll(Arrays.asList(aliases));
+    }
+
+    protected void addNetworkAlias(String alias) {
+        this.networkAliases.add(alias);
+    }
+
+    protected void setNetworkMode(String networkMode) {
+        this.networkMode = networkMode;
+    }
+
+    protected void setPrivilegedMode(boolean privilegedMode) {
+        this.privilegedMode = privilegedMode;
+    }
+
+    public List<Bind> getBinds() {
+        return new ArrayList<>(this.binds);
+    }
+
+    protected void setBinds(List<Bind> binds) {
+        this.binds.clear();
+        this.binds.addAll(binds);
+    }
+
+    protected void addBinds(Bind... binds) {
+        this.binds.addAll(Arrays.asList(binds));
+    }
+
+    protected void addBind(Bind bind) {
+        this.binds.add(bind);
+    }
+
+    protected void setWaitStrategy(WaitStrategy waitStrategy) {
+        this.waitStrategy = waitStrategy;
+    }
+
+    static class Started implements StartedContainer {
+
+        @Delegate(types = ContainerState.class)
+        protected final ContainerState containerState;
+
+        public Started(ContainerState containerState) {
+            this.containerState = containerState;
+        }
+
+        @Override
+        public void close() {
+            if (this.containerState instanceof GenericContainer) {
+                ((GenericContainer<?>) this.containerState).stop();
+                return;
+            }
+            //        if (this instanceof ContainerHooksAware) {
+            //            ((ContainerHooksAware) this).containerIsStopping();
+            //        }
+            //        ResourceReaper.instance().stopAndRemoveContainer(getContainerId());
+            //        if (this instanceof ContainerHooksAware) {
+            //            ((ContainerHooksAware) this).containerIsStopped();
+            //        }
```

Comment:
Those can be removed but rather to wait for https://github.com/testcontainers/testcontainers-java/pull/7782#discussion_r1387198617